### PR TITLE
plugins: add support for plugin configs

### PR DESCRIPTION
**Pull request by Stebalien**
For now, configs specified in `daemon --init-config` and `init CONFIG` are not available. We should fix this eventually but isn't necessary for now (and supporting this will be annoying).

### CODE REVIEW

**Comment by Stebalien** [comment]
@djdv CI is passing. This should be ready for review.

File: plugin/loader/loader.go [hunk 1]
```diff
@@ -83,16 +86,32 @@ type PluginLoader struct {
...
 	plugins map[string]plugin.Plugin
 	started []plugin.Plugin
+	config  config.Plugins
+	repo    string
 }
 
 // NewPluginLoader creates new plugin loader
-func NewPluginLoader() (*PluginLoader, error) {
-	loader := &PluginLoader{plugins: make(map[string]plugin.Plugin, len(preloadPlugins))}
+func NewPluginLoader(repo string) (*PluginLoader, error) {
```
Comment (djdv) [inline plugin/loader/loader.go]: The repo a Plugin receives from the Environment should probably be normalized into absolute paths.
Not sure though, there may be a case for accepting relatives.

Abs patch 1/2
```suggestion
func NewPluginLoader(repo string) (*PluginLoader, error) {
	var err error
	if repo, err = filepath.Abs(repo); err != nil {
		return nil, err
	}
```

File: plugin/loader/loader.go [hunk 2]
```diff
@@ -83,16 +86,32 @@ type PluginLoader struct {
...
+			return nil, err
+		}
+	}
 	for _, v := range preloadPlugins {
 		if err := loader.Load(v); err != nil {
 			return nil, err
 		}
 	}
+
+	if err := loader.LoadDirectory(filepath.Join(repo, "plugins")); err != nil {
```
Comment (djdv) [inline plugin/loader/loader.go]: Abs patch 2/2
```suggestion
	if err = loader.LoadDirectory(filepath.Join(repo, "plugins")); err != nil {
```

Comment (Stebalien) [reply plugin/loader/loader.go]: In general, I don't like using `=` in assignments inside if statements. The point of using an if statement like this is to scope the `err` variable to the if statement. Using a normal assignment lets it escape the scope and can be really confusing.

Comment (Stebalien) [reply plugin/loader/loader.go]: Basically, this should either be left as-is or should be:

```
err = loader.LoadDirectory(...)
if err != nil {
  // ...
}
```

File: plugin/loader/loader.go (hunk 1 above)
Comment (Stebalien) [reply plugin/loader/loader.go]: Given that we don't do this anywhere else, I'd like to leave this as it is for consistency (that way the repo path is the same everywhere). I believe it'll _usually_ be an absolute path anyways.

If we do run into issues, we should be consistent and make this path absolute at a higher layer.

**Pull request merged**